sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/UserDefinedFunction.scala

```diff
@@ -86,6 +86,11 @@ object UserDefinedFunction {
   // The default Hive Metastore SQL schema length for function resource uri.
   private val HIVE_FUNCTION_RESOURCE_URI_LENGTH_THRESHOLD: Int = 4000
 
+  def parseRoutineParam(text: String, parser: ParserInterface): StructType = {
```

Comment:
it seems this method is exactly the same as `parseTableSchema`?

Reply:
parseTableSchema parses a colTypeList rather than a colDefinitionList. colDefinition accepts options like DEFAULT and GENERATED ALWAYS AS but colType doesn't